binja/main.py

```diff
@@ -0,0 +1,62 @@
+import os
+import subprocess
+import sys
+from shutil import copy
+import tarfile
+import json
+
+files_set = set()
+curdir = os.getcwd()
+
+# Handle an individual file
+def analyze(exe, core_file):
+    global files_set
+    pygdb = curdir + "/gdb.py"
+    lines = (
+        subprocess.check_output(
+            ["gdb", exe, core_file, "-batch", "-ex", "source " + pygdb, "--quiet"]
+        )
+        .decode("utf-8")
+        .split("\n")[4:]
+    )
+
+    files = []
+    for l in lines:
+        a = l.split()
+        if len(a) != 2:
+            continue
+        addr = a[0]
+        file = a[1]
+        files_set.add(file)
+        file = file.split("/")[-1]
+        if len(files) == 0 or files[-1]["file"] != file:
+            files.append({"file": file, "stack": []})
+        files[-1]["stack"].append(addr)
+
+    return files
+
+
+if __name__ == "__main__":
```

Comment:
Do we currently handle deduping/deconflicting of filenames anywhere? We should probably make sure that a given analysis run doesn't clobber (or accidentally re-use) the files of a previous run.

Reply:
What do you mean by this?

The only file operations done are that it creates a new krfanalysis directory and `krfanalysis.tar.gz` file each time, and if the directory already exists it fails.

Reply:
> The only file operations done are that it creates a new krfanalysis directory and `krfanalysis.tar.gz` file each time, and if the directory already exists it fails.

I was thinking something like `krfanalysis-{programname}-{timestamp}.tar.gz` -- that way you could run the analysis multiple times on a single input and easily check for discrepancies, or run multiple different analyses in parallel without having to sync on the filename.